build/source/engine/layerDivide.f90

```diff
@@ -259,17 +260,24 @@ subroutine layerDivide(&
 
   ! identify the number of layers to check for need for sub-division
   select case(ix_snowLayers)
-   case(sameRulesAllLayers);    nCheck = nSnow
-   case(rulesDependLayerIndex); nCheck = min(nSnow,4)  ! the depth of the 5th layer, if it exists, does not have a maximum value
+   case(sameRulesAllLayers);    nCheck = min(nSnow, maxSnowLayers-1)
+   case(rulesDependLayerIndex); nCheck = min(nSnow, maxSnowLayers-1)  ! the depth of the 5th layer, if it exists, does not have a maximum value
    case default; err=20; message=trim(message)//'unable to identify option to combine/sub-divide snow layers'; return
   end select ! (option to combine/sub-divide snow layers)
 
   ! loop through all layers, and sub-divide a given layer, if necessary
   do iLayer=1,nCheck
+   divideLayer=.false.
 
    ! identify the maximum depth of the layer
    select case(ix_snowLayers)
-    case(sameRulesAllLayers);    zmaxCheck = zmax
+    case(sameRulesAllLayers)
+     if (nCheck >= 99) then
```

Comment:
What are the magic numbers `99` and two lines laters `99999.`?

Reply:
99 was a standin for `maxSnowLayers-1`. The `jrdn1991` snow layering decision sets `maxSnowLayers` to 100. So, the aim is to never split the snowpack into more than `maxSnowLayers`.